test/unit/classDecorator.spec.ts

```diff
@@ -0,0 +1,161 @@
+    const source = `
+    function SetNumbers(this: void, ...numArgs: number[]) {
+        return <T extends new(...args: any[]) => {}>(constructor: T) => {
+            return class extends constructor {
+                decoratorNums = new Set<number>(numArgs);
+            };
+        };
+    }
+
+    @SetNumbers(120, 30, 54)
+    class TestClass {
+        public decoratorNums;
+    }
+
+    const classInstance = new TestClass();
+    let sum = 0;
+    for (const value of classInstance.decoratorNums) {
+        sum += value;
+    }
+    return sum;
+    `;
+
+    const result = util.transpileAndExecute(source);
+    expect(result).toBe(204);
+});
+
+test("Multiple class decorators", () => {
+    const source = `
+    function SetTen<T extends { new(...args: any[]): {} }>(constructor: T) {
+        return class extends constructor {
+            decoratorTen = 10;
+        }
+    }
+
+    function SetNum(this: void, numArg: number) {
+        return <T extends new(...args: any[]) => {}>(constructor: T) => {
+            return class extends constructor {
+                decoratorNum = numArg;
+            };
+        };
+    }
+
+    @SetTen
+    @SetNum(410)
+    class TestClass {
+        public decoratorTen;
+        public decoratorNum;
+    }
+
+    const classInstance = new TestClass();
+    return classInstance.decoratorNum + classInstance.decoratorTen;
+    `;
+
+    const result = util.transpileAndExecute(source);
+    expect(result).toBe(420);
+});
+
+test("Class decorator with inheritance", () => {
+    const source = `
+    function SetTen<T extends { new(...args: any[]): {} }>(constructor: T) {
+        return class extends constructor {
+            decoratorTen = 10;
+        }
+    }
+
+    function SetNum(this: void, numArg: number) {
+        return <T extends new(...args: any[]) => {}>(constructor: T) => {
+            return class extends constructor {
+                decoratorNum = numArg;
+            };
+        };
+    }
+
+    class TestClass {
+        public decoratorTen = 0;
+        public decoratorNum = 0;
+    }
+
+    @SetTen
+    @SetNum(410)
+    class SubTestClass extends TestClass {}
+
+    const classInstance = new SubTestClass();
+    return classInstance.decoratorNum + classInstance.decoratorTen;
+    `;
+
+    const result = util.transpileAndExecute(source);
+    expect(result).toBe(420);
+});
+
+test("Class decorators are applied in reverse order", () => {
+    const source = `
+    function SetString(this: void, stringArg: string) {
+        return <T extends new(...args: any[]) => {}>(constructor: T) => {
+            return class extends constructor {
+                decoratorString = stringArg;
+            };
+        };
+    }
+
+    @SetString("the quick brown fox")
+    @SetString("jumped")
+    @SetString("over the lazy dog")
+    class TestClass {
+        public decoratorString = "";
+    }
+
+    const classInstance = new TestClass();
+    return classInstance.decoratorString;
+    `;
```

Comment:
I think this test can be improved since it only really shows the result of one of the decorators. How about something like this:
```ts
const order = [];

function SetString(this: void, stringArg: string) {
    order.push("eval " + stringArg);
    return <T extends new (...args: any[]) => {}>(constructor: T) => {
        order.push("execute " + stringArg);
        return class extends constructor {
            decoratorString = stringArg;
        };
    };
}

@SetString("the quick brown fox")
@SetString("jumped")
@SetString("over the lazy dog")
class TestClass {
    public static decoratorString = "";
}

const inst = new TestClass();
return JSONStringify(order);
```